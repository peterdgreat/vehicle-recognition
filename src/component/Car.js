import React , {useState} from 'react'
import ImageUploading from 'react-images-uploading';
import { useDispatch, useSelector } from 'react-redux';
import   { getVehicle } from '../redux/Vehicles/vehicle';
import { NavLink } from 'react-router-dom';
import {useNavigate} from 'react-router-dom';

export default function Car(props) {
    const dispatch = useDispatch();
 const [images, setImages] = useState([]);
  const maxNumber = 1;

  const onChange = (imageList, addUpdateIndex) => {
    // data for submit
    console.log(imageList, addUpdateIndex);
    setImages(imageList);
  };
const navigate = useNavigate();

  const onUpload = () => {
      if (images.length ===1) {
          navigate('/details');
        dispatch(getVehicle());
     
      }
  };

  return (
    <div className="App">

      <ImageUploading
        multiple= "false"
        value={images}
        onChange={onChange}
        maxNumber={maxNumber}
        dataURLKey="data_url"
      >
        {({
          imageList,
          onImageUpload,
          onImageUpdate,
          isDragging,
          dragProps,
        }) => (
      
          <div className="upload__image-wrapper">
            <button
              style={isDragging ? { color: 'red' } : undefined}
              onClick={onImageUpload}
              {...dragProps}
            >
            Drag or Upload a Car Image
            </button>
            {imageList.map((image, index) => (
              <div key={index} className="image-item">
                <img src={image['data_url']} alt="" width="100" />
                <div className="image-item__btn-wrapper">
                  <button onClick={() => onImageUpdate(index)}>Update</button>
                    <button  onClick= {onUpload}>Upload</button>
                </div>
              </div>
            ))}
          </div>
        )}
      </ImageUploading>
 

      </div>
  
  );
}
