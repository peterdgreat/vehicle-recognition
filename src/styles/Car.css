.card {
    background-color: transparent;
}

.img-uploaded {
    width: 70%;
    object-fit: contain;
    height: 70%;
    border: 1px solid #fff;
    margin: 0 5px 0 5px;

}

.btn {
    color: rgba(255, 255, 255, 0.767);
    background-color: transparent;
}

.btn:hover {
    color: #fff;
    transition: all 1s ease-in-out;
}

.sp-btn:hover {
    color: #1c1c1eec;
    background-color: #fff;
    transition: all 1s ease-in-out;
}
