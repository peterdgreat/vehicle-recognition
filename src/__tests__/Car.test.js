import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import Car from '../component/Car';
import store from '../redux/configurestore';

describe('Car', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
       <Router>
        <Routes>
           <Route path="/" element={<Car />} />
        </Routes>
         </Router>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});