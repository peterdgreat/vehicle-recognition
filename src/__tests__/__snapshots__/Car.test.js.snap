// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Car To Match Snapshot renders correctly 1`] = `
<div
  className="App"
>
  <input
    accept="image/*"
    multiple={true}
    onChange={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />
  <div
    className=""
  >
    <div
      className="card"
    >
      <button
        className="btn border "
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
      >
        <h1>
          Drag or Upload a Car Image
        </h1>
      </button>
    </div>
  </div>
</div>
`;
